common/analysis/lint_rule_status.cc

```diff
@@ -82,11 +82,33 @@ void LintStatusFormatter::FormatLintRuleStatuses(
       violations.insert(LintViolationWithStatus(&violation, &status));
     }
   }
+  return violations;
+}
+
+void LintStatusFormatter::FormatLintRuleStatuses(
+    std::ostream* stream, const std::vector<LintRuleStatus>& statuses,
+    absl::string_view base, absl::string_view path) const {
+  auto violations = PrepareViolations(statuses);
+  for (auto violation : violations) {
+    FormatViolation(stream, *violation.violation, base, path,
+                    violation.status->url, violation.status->lint_rule_name);
+    *stream << std::endl;
+  }
+}
+
+void LintStatusFormatter::FormatLintRuleStatuses(
```

Comment:
This duplication of functions is of course increasing the maintenance burden in the future.

I'd prefer it to be only one function.

Maybe we remove the old function and add this new function. And we would only ...